sdk/js/src/index.ts

```diff
@@ -105,7 +105,7 @@ export function initializeDevCycle<
         | DevCycleOptionsWithDeferredInitialization,
     optionsArg: DevCycleOptions = {},
 ): DevCycleClient<Variables, CustomData> {
-    if (!sdkKey) {
+    if (!sdkKey && optionsArg?.throwOnError) {
```

Comment:
can we change this to have an IF statement inside which will at least `this.logger.error` instead of failing silently?

Also the `throwOnError` should default to true as that is our existing behaviour.
 
```suggestion
    if (!sdkKey) {
      if (optionsArg?.throwOnError) {
        throw new UserError(
            'Missing SDK key! Call initialize with a valid SDK key',
        )
      } else {
        this.logger.error('Missing SDK key! Call initialize with a valid SDK key)
      }
    }
```